src/main/java/gregtech/api/gui/widgets/TextFieldWidget2.java

```diff
@@ -129,13 +148,17 @@ public void drawInBackground(int mouseX, int mouseY, float partialTicks, IRender
         }
         if (focused && drawCursor) {
             // render cursor
-            String sub = text.substring(0, cursorPos);
+            String sub = text.substring(0, toRenderTextIndex(cursorPos));
             float x = fontRenderer.getStringWidth(sub) * scale + textX;
             x *= scaleFactor;
             drawCursor(x, y);
         }
         GlStateManager.popMatrix();
         GlStateManager.enableBlend();
+
+        if (this.unicodeMode != null) {
+            fontRenderer.setUnicodeFlag(prevUnicode);
+        }
```

Comment:
The code overrides "force unicode" option in font renderer and reverts the value to previous value once the operation is over. I don't think there's better way of writing this bit.

This was added as a part of ore filter UI overhaul, but ended up being unused. Should I just remove it? It does work, and is a trivial addition to feature set, but it's unlikely that it would be used anywhere.